DOCS-629 Chef docs (#701)
DataDog/chef-datadog: CONTRIBUTING.md

# Contributing

The code is licensed under the Apache License 2.0 (see  LICENSE for details).

[![Chef cookbook](https://img.shields.io/cookbook/v/datadog.svg?style=flat)](https://github.com/DataDog/chef-datadog)
[![Build Status](https://img.shields.io/circleci/build/gh/DataDog/chef-datadog.svg)](https://circleci.com/gh/DataDog/chef-datadog)
[![GitHub forks](https://img.shields.io/github/forks/DataDog/chef-datadog.svg)](https://github.com/DataDog/chef-datadog/network)
[![GitHub stars](https://img.shields.io/github/stars/DataDog/chef-datadog.svg)](https://github.com/DataDog/chef-datadog/stargazers)

First of all, thanks for contributing!

This document provides guidelines and instructions for contributing to this repository. To propose improvements, feel free to submit a PR.

## Submitting issues

* If you think you've found an issue, search the issue list to see if there's an existing issue.
* Then, if you find nothing, open a Github issue.

## Pull Requests

Have you fixed a bug or written a new feature and want to share it? Many thanks!

> In order to ease/speed up our review, here are some items you can check/improve when submitting your PR:

Reviewer: ```suggestion
When submitting your PR, here are some items you can check or improve to facilitate the review process:
```

Kitchen allows you to test specific recipes described in [kitchen.yml](./kitchen.yml) across all platforms and versions that are also described in the same file. Each combination of recipe x platform x version is a test target.

To list available targets, use the `list` command:

```bash
bundle exec kitchen list
```

To test a specific target, run:

```bash
bundle exec kitchen test <target>
```

For example, to test the CouchDB monitor on Ubuntu 16.04, run:

```bash
bundle exec kitchen test datadog-couchdb-ubuntu-1604-15
```

There is a CouchDB recipe described in the `kitchen.yml`, and an Ubuntu platform with the 16.04 version.

### Development loop

The simplest way to develop fixes or features is to set up the platform and version of your choice with the `create` command and apply the recipe with the `converge` command. To explore the machine and test different things, login with the `login` command.

**Note**: The `login` command only works on Linux and OSX. For Windows, connect to the VM through the Virtual Box interface or with putty or a similar ssh client.

> N.B.: The credentials of the created virtual machines are login `vagrant`, password `vagrant`.

Reviewer 2: ```suggestion
**Note**: The credentials of the created virtual machines are login `vagrant`, password `vagrant`.
```
